go/ct/common/hashes256_test.go

```diff
@@ -0,0 +1,114 @@
+//
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the GNU Lesser General Public Licence v3
+//
+
+package common
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/vm"
+)
+
+func TestImmutableHashArray_EqualWhenContainingSameContent(t *testing.T) {
+	b1 := NewImmutableHashArray(vm.Hash{1})
+	b2 := b1
+	b3 := NewImmutableHashArray(vm.Hash{2})
+
+	if !b1.Equal(b2) {
```

Comment:
The test is named "EqualWhenContainngSameContent" but what is tested is only the case where the content is the same pointer. You should also make sure that two independent instances created by two distinct calls to `NewImmutableHashArray` with the same arguments are equal.